src/vklayer.c

```diff
@@ -1138,13 +1153,17 @@ static void vk_shtex_capture(struct vk_data *data,
     VkSubmitInfo submit_info;
     submit_info.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
     submit_info.pNext = NULL;
-    submit_info.waitSemaphoreCount = 0;
-    submit_info.pWaitSemaphores = NULL;
+    submit_info.waitSemaphoreCount = info->waitSemaphoreCount;
+    submit_info.pWaitSemaphores = info->pWaitSemaphores;
     submit_info.pWaitDstStageMask = NULL;
```

Comment:
This cannot be NULL now.

Reply:
I don't think the VkPresentInfoKHR* ever would be null, since in correct API usage one must be passed in.

See: https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkQueuePresentKHR.html

I can't see anywhere where this gets called outside of vkQueuePresentKHR, were there some expected changes in that would rely on this being NULL? or are you pointing me to do something else?

Reply:
I meant pWaitDstStageMask:

>  If waitSemaphoreCount is not 0, pWaitDstStageMask must be a valid pointer to an array of waitSemaphoreCount valid combinations of VkPipelineStageFlagBits values

Reply:
So it would be

```cpp
    submit_info.pWaitDstStageMask = info->pWaitDstStageMask;

    VkPipelineStageFlagBits waitStage = VK_PIPELINE_STAGE_TRANSFER_BIT; // ?
    info->pWaitDstStageMask = &waitStage;
```

Reply:
Okay, so pWaitDstStageMask needs to be an array, one for each semaphore we're waiting on.

I didn't want to introduce allocation for each frame, so I've settled on a static list.

See the commit message 860cd11b4f8cb4b013832074906159fafd022742 for details

Reply:
Yeah, let's not complicate it too much :D

Just make it a static array and initialize it in `OBS_Negotiate` (either with memset, or for loop).

Reply:
Okay, will do.

Reply:
Done